Tools/WinMLRunner/src/Run.cpp

```diff
@@ -799,39 +831,48 @@ int run(CommandLineArgs& args, Profiler<WINML_MODEL_TEST_PERF>& profiler) try
 #endif
                     LearningModelSession session = nullptr;
                     IDirect3DDevice winrtDevice = nullptr;
-                    lastHr = CreateSession(session, winrtDevice, model, args, output, deviceType,
-                                           deviceCreationLocation, profiler);
-                    if (FAILED(lastHr))
-                    {
-                        continue;
-                    }
                     for (auto inputDataType : inputDataTypes)
                     {
                         for (auto inputBindingType : inputBindingTypes)
                         {
-                            // Clear up bind, eval performance metrics after iteration
+                            // Clear up session, bind, eval performance metrics after configuration iteration
                             if (args.IsPerformanceCapture() || args.IsPerIterationCapture())
                             {
                                 // Resets all values from profiler for bind and evaluate.
                                 profiler.Reset(WINML_MODEL_TEST_PERF::BIND_VALUE, WINML_MODEL_TEST_PERF::COUNT);
                             }
-                            if (args.IsImageInput())
+                            for (uint32_t sessionCreationIteration = 0;
+                                sessionCreationIteration < args.NumSessionCreationIterations();
+                                sessionCreationIteration++)
                             {
-                                for (const std::wstring& inputImagePath : args.ImagePaths())
+                                lastHr = CreateSession(session, winrtDevice, model, args, output, deviceType,
```

Comment:
As we have discussed, it's probably better to keep CreateSession outside of this nested for loops, and in the function CreateSession, you can create a session, load and eval for each input types